UnityClient/Assets/Scripts/PatternCore/Generics/SingletonDontDestroy.cs

```diff
@@ -0,0 +1,18 @@
+﻿using UnityEngine;
+
+public class SingletonDontDestroy<T> : MonoBehaviour where T : MonoBehaviour
```

Comment:
I think `SingletonMonoBehaviour` would be a better name

Reply:
Name changed